portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,59 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("Login")
+public class LoginServlet extends HttpServlet {
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        UserService userService = UserServiceFactory.getUserService();
+        PrintWriter out = response.getWriter();
+
+        if (!userService.isUserLoggedIn()) {
+            String loginUrl = userService.createLoginURL("/Login");
+            out.println("<p>Login <a href=\"" + loginUrl + "\">here</a>.</p>");
+        } else {
+            String logoutUrl = userService.createLogoutURL("/");
+            out.println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+            out.println("<p>Username:");
+            out.println("<form method=\"POST\" action=\"Login\">");
+            out.println("<input name=\"username\" >");
+            out.println("<br />");
+            out.println("<button>Submit</button>");
+            out.println("</form>");
+        }
+    }
+
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        UserService userService = UserServiceFactory.getUserService();
+        if (!userService.isUserLoggedIn()) {
+            response.sendRedirect("/Login");
```

Comment:
I'm not 100% sure on this, but I thiiiink it is a good practice to be consistent in capitalization convention with your URLs.  And also I believe all lowercase is preferred (and hyphens, if there are multiple words).  I see you have other URLs/endpoints that are in all lowercase already.. I would probably lowercase "/login" as well unless you had a specific reason to break from the general conventions (that I think exist..).

Reply:
Lowercased login